Packet++/header/TcpLayer.h

```diff
@@ -439,12 +449,12 @@ namespace pcpp
 		 * Add a new TCP option after an existing one
 		 * @param[in] optionBuilder A TcpOptionBuilder object that contains the requested TCP option data to be added
 		 * @param[in] prevOptionType The TCP option which the newly added option should come after. This is an optional parameter which
-		 * gets a default value of ::TCPOPT_Unknown if omitted, which means the new option will be added as the first option in the layer
+		 * gets a default value of TcpOptionType::TCPOPT_Unknown if omitted, which means the new option will be added as the first option in the layer
 		 * @return A TcpOption object containing the newly added TCP option data or logical NULL
 		 * (TcpOption#isNull() == true) if addition failed. In case of a failure a corresponding error message will be
 		 * printed to log
 		 */
-		TcpOption addTcpOptionAfter(const TcpOptionBuilder& optionBuilder, TcpOptionType prevOptionType = TCPOPT_Unknown);
+		TcpOption addTcpOptionAfter(const TcpOptionBuilder& optionBuilder, TcpOptionType prevOptionType = TcpOptionType::TCPOPT_Unknown);
```

Comment:
we just break the backward compacity here.
probably we should keep both the old  `enum TcpOptionType` and the new `enum class TcpOptionType` at least for now.
Have an overload function for both the old enum and the new enum class.
The one which takes the old enum way should add `PCPP_DEPRECATED`.
```cpp
TcpOption addTcpOptionAfter(const TcpOptionBuilder& optionBuilder, TcpOptionTypeNew prevOptionType = TcpOptionTypeNew::TCPOPT_Unknown);
 
PCPP_DEPRECATED TcpOption addTcpOptionAfter(const TcpOptionBuilder& optionBuilder, TcpOptionType prevOptionType = TCPOPT_Unknown);
```
free feel to have another better approach.

Reply:
Unless I am missing something I am afraid the solution with macros is not an option.

If we make a macro: `#define PCPP_TCPOPT_EOL TcpOptionType::PCP_TCPOP_EOL`

And then try to use strongly typed version in client code:

```
switch (option_type) {
    case TcpOptionType::PCPP_TCPOPT_EOL:
      ....
}
```
case will be evaluated to: `case TcpOptionType::TcpOptionType` and not compile.